q2cli/core/config.py

```diff
@@ -0,0 +1,99 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2019, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import os
+import configparser
+
+import click
+
+import q2cli.util
+
+
+class CLIConfig():
+    path = os.path.join(q2cli.util.get_app_dir(), 'cli-colors.theme')
+    VALID_SELECTORS = {'option', 'type', 'default_arg', 'command', 'emphasis',
+                       'problem', 'error', 'required', 'success'}
+    VALID_STYLINGS = {'fg', 'bg', 'bold', 'dim', 'underline', 'blink',
+                      'reverse'}
+    VALID_COLORS = {'black', 'red', 'green', 'yellow', 'blue', 'magenta',
+                    'cyan', 'white', 'bright_black', 'bright_red',
+                    'bright_green', 'bright_yellow', 'bright_blue',
+                    'bright_magenta', 'bright_cyan', 'bright_white'}
+    VALID_BOOLEANS = {'true': True,
+                      'false': False,
+                      't': True,
+                      'f': False}
+
+    def __init__(self):
+        self.styles = self._get_default_styles()
+        try:
+            self.parse_file(self.path)
+        except Exception:
+            if os.path.exists(self.path):
+                os.unlink(self.path)
+            self.styles = self._get_default_styles()
+
+    def _get_default_styles(self):
+        return {'option': {'fg': 'blue'},
+                'type': {'fg': 'green'},
+                'default_arg': {'fg': 'magenta'},
+                'command': {'fg': 'blue'},
+                'emphasis': {'underline': True},
+                'problem': {'fg': 'yellow'},
+                'error': {'fg': 'red', 'bold': True},
+                'required': {'underline': True},
+                'success': {'fg': 'green'}}
+
+    def _build_error(self, current, valid_list, valid_string):
+        valids = ''
+        for valid in valid_list:
+            valids += valid + '\n'
```

Comment:
These three lines can become just:
```
'\n'.join(valid_list)
```
That will make an empty string for an empty list, and only put newlines between the lines (saving you the double new line at the end from the final echo).